src/care/PluginData.h

```diff
@@ -0,0 +1,65 @@
+#ifndef _CARE_PluginData_H_
+#define _CARE_PluginData_H_
+
+#include "chai/ExecutionSpaces.hpp"
+#include "care/config.h"
+#include "care/CHAICallback.h"
+#include <vector>
+
+namespace chai {
+   struct PointerRecord;
+}
+
+namespace care{
+
+	class PluginData	{
+		public:
+			PluginData();
+
+         static void setFileName(const char * name);
+
+         static void setLineNumber(int num);	
+
+         static const char * getFileName();
+         
+         static int getLineNumber();   
+
+         static void setParallelContext(bool isParallel);
+
+         static bool isParallelContext();
+
+         static bool post_parallel_forall_action_registered(void * key); 
+
+         static void register_post_parallel_forall_action(void * key, std::function<void(chai::ExecutionSpace, const char *, int)> action);
+
+         static void setSynchronization(bool synchronizeBefore, bool synchronizeAfter);
```

Comment:
`setSynchronization` doesn't appear to do anything any more. I think it should probably be removed.

Reply:
The variables that it sets can probably be removed, too, and maybe some things that call setSynchronization, too.